src/files/file-preview/FilePreview.js

```diff
@@ -27,8 +29,8 @@ const PreviewItem = ({ t, name, cid, size, type, availableGatewayUrl: gatewayUrl
   const [content, setContent] = useState(null)
 
   const loadContent = async () => {
-    const buf = await read()
-    setContent(buf.toString('utf-8'))
+    const buf = await first(await read())
```

Comment:
Hm.. this will only show single block. To show full file you need to use [concat](https://www.npmjs.com/package/it-concat) or we need some UI to indicate truncation.

Example: try previewing text file that is bigger than a single block, it will get truncated:

- https://www.gutenberg.org/files/2600/2600-0.txt

> ![2020-09-18--18-23-39](https://user-images.githubusercontent.com/157609/93621649-2bbab880-f9dc-11ea-9734-4a8b002c0adc.png)

Reply:
I'd vote for truncation (if file was bigger than 256 KB, add footer with link to full file), as its easier on resources, but will lean on @jessicaschilling to make a call what is the better fix here.

Reply:
Totally agree with @lidel.

Reply:
Thank you, didn't thought of testing big text files 😢 


Changed to concat, should be 👌 now